### ISSUE TITLE
Training the model on own dataset

### ISSUE THREAD

**Issue by Andre1998Shuvam** [eladrich/pixel2style2pixel#119]
Hello,
I have been trying to convert old Renaissance portraits to modern realistic photographs, i.e., realify the portrait faces. I have created a custom dataset for this and for the training purpose, I have set both the train_source_root and train_target_root as "portrait_train", i.e., the name for my training dataset. I have used "ffhq_encode" transformation for my images and the pretrained stylegan-config-ffhq.pt model. 

However, after considerable training iterations, I am finding the output images not resembling the original ones in many cases and there is low diversity in the generated faces. I have also tried changing the loss lambda parameters but in vain. 
I have changed the id_lambda to 1.5 and the l2_lambda from 1 to 0.5, as I thought that would decrease the pixel to pixel similarity yet maintaining the facial identity with the original.

It will be very helpful if you can kindly guide us in this context.

**Comment by yuval-alaluf** [comment]
Hi @Andre1998Shuvam,
For us to help you, please provide an example of what kind of results you're getting on the train and test (examples of the `logs` outputs).
Also, please provide the parameters you used (e.g., the batch size).
For now, please note that an id lambda of 1.5 seems quite large (we typically use a lambda of 0.1).

**Comment by Andre1998Shuvam** [comment]
Hello!
Here are two sample results:

![9000](https://user-images.githubusercontent.com/49407884/114412782-0afdbe80-9bcb-11eb-87ea-987d817c0850.jpg)
![8200](https://user-images.githubusercontent.com/49407884/114413284-78a9ea80-9bcb-11eb-97b4-c2ec2ff6bbc7.jpg)

The parameters I have used:
![Screenshot from 2021-04-12 20-14-04](https://user-images.githubusercontent.com/49407884/114413527-ae4ed380-9bcb-11eb-9225-d478ddfd4258.png)

I have initially used id_lambda as 1 as you have mentioned while doing the Toonification process using your model based on an unpaired dataset. However, after 10000-12000 iterations, I noticed that the output images are becoming exactly similar to the target images, i.e., they are no longer remaining realistic and instead, getting converted back to the Renaissance Portrait style images. That's why, I thought of increasing the id_lambda and on the other hand, decreasing the l2_lambda to reduce the pixel to pixel similarity to prevent the output images from becoming Renaisance portrait-like.

**Comment by yuval-alaluf** [comment]
So a couple of things: 
1. In all of our experiments, we used a batch size of 8 and the loss weights were adjusted to accommodate this (e.g., the LPIPS loss is 0.8 or 0.1 per sample). Therefore, you should first try scaling your losses accordingly. 
2. Since the outputs are frontalized, it seems like reducing the L2 loss had a detrimental effect as you are unable to capture the input pose correctly.
3.  You wrote that you tried to "reduce the pixel to pixel similarity to prevent the output images from becoming Renaisance portrait-like."
If you wish to remain closer to the "realistic" faces, you may need to push the latents closer to the latent space that StyleGAN was trained on. You could try doing this by increasing the `w_norm_lambda`, but I am unsure if this will be enough. 

Overall, it could very well be that pSp, as is, may not be enough for solving your task. You may want to consider incorporating a cycle-based component into your training (in an unpaired setting similar to CycleGAN). For example, you'll want to try to translate the painting to real faces and then back to paintings (i.e., A -> B -> A).
